cmdfactory/builder.go

```diff
@@ -478,6 +478,28 @@ func AttributeFlags(c *cobra.Command, obj any, args ...string) error {
 	return nil
 }
 
+// FetchAlternateConfigDirectory returns the path to the alternate config directory
+// that can be set via the --config-dir flag. This needs to be fetched before flags
+// are populated with AttributeFlags to ensure that the function is called only once.
+func FetchAlternateConfigDirectory(args []string) (path string) {
+	for idx, arg := range args {
+		if strings.HasPrefix(arg, "--config-dir") {
+			continue
+		}
+		if strings.Contains(arg, "=") {
+			if split := strings.Split(arg, "="); len(split) == 2 {
+				path = split[1]
+			}
+		} else {
+			if !strings.HasPrefix(args[idx+1], "-") {
+				path = args[idx+1]
+			}
+		}
+		break
+	}
+	return
+}
+
```

Comment:
Since this is KraftKit-specific, i'd migrate it into `kraftkit.sh/config` as this is where the definition of `--config-dir` exists as well.

Additionally, since this is parsing args and to be more explicit in its method name, let's renamed it to `FetchAlternateConfigDirFromArgs`.

Reply:
Hope the file is correct :crossed_fingers: